src/gh/components/DF_joint_segmentator/metadata.json

```diff
@@ -0,0 +1,92 @@
+{
+    "name": "Extract Joints",
+    "nickname": "Extract Joints",
+    "category": "diffCheck",
+    "subcategory": "Segmentation",
+    "description": "Extracts the joints from a point cloud.",
+    "exposure": 4,
+    "instanceGuid": "fd804928-bf11-43eb-a29f-8afabefa660d",
+    "ghpython": {
+        "hideOutput": true,
+        "hideInput": true,
+        "isAdvancedMode": true,
+        "marshalOutGuids": true,
+        "iconDisplay": 2,
+        "inputParameters": [
+            {
+                "name": "i_recompute",
+                "nickname": "i_recompute",
+                "description": "Connect a button to recompute the registration.",
+                "optional": true,
+                "allowListAccess": false,
+                "showTypeHints": true,
+                "scriptParamAccess": "item",
+                "wireDisplay": "default",
+                "sourceCount": 0,
+                "typeHintID": "bool"
+            },
+            {
+                "name": "i_clusters",
+                "nickname": "i_clusters",
+                "description": "The cluster point clouds from normal segmentation.",
+                "optional": false,
+                "allowTreeAccess": false,
+                "showTypeHints": true,
+                "scriptParamAccess": "list",
+                "wireDisplay": "default",
+                "sourceCount": 0,
+                "typeHintID": "pointcloud"
+            },
+            {
+                "name": "i_joint_faces",
+                "nickname": "i_joints",
+                "description": "The joints to extract and analyze",
+                "optional": false,
+                "allowTreeAccess": false,
+                "showTypeHints": true,
+                "scriptParamAccess": "list",
+                "wireDisplay": "default",
+                "sourceCount": 0,
+                "typeHintID": "mesh"
+            },
+            {
+                "name": "i_joint_ids",
+                "nickname": "i_joint_ids",
+                "description": "The joint ids of the assembly",
+                "optional": false,
+                "allowTreeAccess": true,
+                "showTypeHints": true,
+                "scriptParamAccess": "list",
+                "wireDisplay": "default",
+                "sourceCount": 0,
+                "typeHintID": "int"
```

Comment:
just for discussion: I still believe that we should stick more and use more our own objects like here `df_beam` that already contains those two elements but I see no problem in keeping it like that for now

Reply:
in that case DFDeconstructBeam becomes a utility "for your information", but isn't needed in a basic workflow ?